aiopvapi/helpers/aiorequest.py

```diff
@@ -38,7 +41,7 @@ async def check_response(response, valid_response_codes):
 class AioRequest:
     """Request class managing hub connection."""
 
-    def __init__(self, hub_ip, loop=None, websession=None, timeout=15):
+    def __init__(self, hub_ip, loop=None, websession=None, timeout=15, api_version=0):
```

Comment:
i dont like the `api_version` defaulting to 0, that doesn't make any sense as a consumer of this class' api. I can see internally that this is handled correctly, but still. Is there an alternative that makes more sense?

Reply:
I meant for 0 to mean not discovered yet, but then would still default to the original behavior prior to having multiple API versions.  If not api_version then discovery did not run yet.

Reply:
@trullock do you have an alertnate suggestion here ?

I am almost done with my refactoring to get this to work with v2 as there were a number of issues to start with - no worries though @alindner19 , hard to get these things right when you dont have the kit to test properly and i would not be surprised if i have broken something v3 here too.

I have adjusteda number of items, including a few things that (from the doco i have at least) i think may have been wrong for v3 and the api. Looks like battery level for instance changed to 0,1,2 in v3 and was 1,2,3 in v1

Ill create a list of things we need tested against v3 too.

I could be working off an old version of the api too - if you have a newer json / yaml feel free to flick it over
```  
openapi: 3.0.3
  info:
    title: PowerView Gen3 Gateway
    description: >-
      The Gateway can be accessed and controlled via HTTP requests over a local
      area network via this API.
    version: 1.12.0
```

Reply:
I wouldnt reject the PR because of this, I cant recall exactly what logic handles the `0` but either make it default to the version it ends up defaulting to after the logic, or if it auto detects then use `None` or something to better indicate an undefined value. 0 is an integer, its conceivabley a valid api version number but we're overloading that to have a special meaning

Reply:
Ive adjusted to default None, which will still default to legacy api in logic - but at least you can tell if 0 was actually specifed as a version or not